lib/nolibc/string.c

```diff
@@ -693,11 +693,9 @@ char *strncat(char *dest, const char *src, size_t n)
 
 	dest = dest + strlen(dest);
 
-	if (src != NULL) {
-		while (n && *src) {
-			n--;
-			*dest++ = *src++;
-		}
+	while (n && *src) {
```

Comment:
Maybe we could check if n is greater then strlen(dest), use max(strlen(dest), n).

Reply:
Hi! Could you please elaborate a little on this to help me understand better? Do you mean to do this instead of using __attribute__((nonnull)) and getting rid of the (src != NULL) check? Or is this an idea for further improvement by replacing the repeated decrementing of n with an iterator that goes from 0 to max(strlen(src), n) - 1?

Reply:
Yes for my initial wrong suggestion. It is only an ideea to check that we don't copy from src array more then length of the string + the null terminator.